tasks/eva_2850/fix_discordant_variants.py

```diff
@@ -108,12 +109,24 @@ def fix_discordant_variants(mongo_source, assembly, rs_file, batch_size=1000):
 
                         # if any rs, present in current batch and scheduled to be processed later, is being merged,
                         # it should be removed from this batch and the rs it merged into should be processed in next batch
-                        if merged_rs is not None and merged_rs in rs_list \
-                                and rs_list.index(merged_rs) > rs_list.index(rs):
-                            logger.info(f"Removing rs {merged_rs} from current batch as it is merged "
+                        if merged_rs is not None and merged_rs in rs_list:
+                            if rs_list.index(merged_rs) > rs_list.index(rs):
+                                logger.info(f"Removing rs {merged_rs} from current batch as it is merged "
                                         f"and putting rs {merge_into} for processing in the next batch")
-                            rs_list.remove(merged_rs)
-                            rs_list_to_process.append(merge_into)
+                                rs_list.remove(merged_rs)
+                                rs_list_to_process.append(merge_into)
+                            else:
+                                del all_rs_variants[merge_into]
+                                all_rs_variants.update(get_rs_variants(mongo_source, assembly, [merge_into]))
+
+                                del dbsnp_ss_variants[merge_into]
+                                del eva_ss_variants[merge_into]
+                                del all_ss_variants[merge_into]
+                                dbsnp_ss, eva_ss, all_ss = get_ss_variants(mongo_source, assembly, rs_list)
+                                dbsnp_ss_variants.update(dbsnp_ss)
+                                eva_ss_variants.update(eva_ss)
+                                all_ss_variants.update(all_ss)
```

Comment:
RS1 and RS2 in same batch (considering RS1 is first in the list), In case of a merge
**if)** If RS2 is merged into RS1and since it is supposed to be processed later, we remove it from the current processing list and put it in the next batch (where the the RS in which it has merged(RS1) will be picked up for processing )
**else)** If RS1 is merged into RS2, it might have caused some changes in the information related to RS2 (all the info  was read at the start of the batch). We are updating all that info, so that when we pick RS2, we have the latest info which will help us in deciding whether to process that or not
For e.g. when we merge RS1 into RS2, all the SS having RS as  RS1 will now have RS RS2 (this was not the case at the start of the batch), we need to fetch CVE and SVE associated with RS2 again to get these changes.

Reply:
have updated the condition and added the reasoning as comment

Reply:
Okay makes sense, thanks!  I agree it would probably be simpler for the else clause to just delete RS2 and include in the next batch, but this is fine.